src/ocean/util/container/map/HashMap.d

```diff
@@ -112,7 +112,7 @@ public class HashMap ( V ) : Map!(V, hash_t)
 
     ***************************************************************************/
 
-    public override hash_t toHash ( hash_t key )
+    public override hash_t toHash ( in hash_t key )
```

Comment:
Good question!

It is result of a bit awkward series of events. Root abstract `toHash` comes from `BucketSet` and is templated on key type. And some keys are non-value types so `const` becomes necessary. But when checking overrides D compiler ensures exact qualifier compatibility even for value types, so you can't override `toHash ( K ) ( in K )` with plain `toHash ( hash_t )`.